### ISSUE TITLE
The Automated Learning of Deep Features for Breast Mass Classification from Mammograms

### ISSUE THREAD

**Issue by cgreene** [greenelab/deep-review#169]
http://doi.org/10.1007/978-3-319-46723-8_13

> The classification of breast masses from mammograms into benign or malignant has been commonly addressed with machine learning classifiers that use as input a large set of hand-crafted features, usually based on general geometrical and texture information. In this paper, we propose a novel deep learning method that automatically learns features based directly on the optmisation of breast mass classification from mammograms, where we target an improved classification performance compared to the approach described above. The novelty of our approach lies in the two-step training process that involves a pre-training based on the learning of a regressor that estimates the values of a large set of hand-crafted features, followed by a fine-tuning stage that learns the breast mass classifier. Using the publicly available INbreast dataset, we show that the proposed method produces better classification results, compared with the machine learning model using hand-crafted features and with deep learning method trained directly for the classification stage without the pre-training stage. We also show that the proposed method produces the current state-of-the-art breast mass classification results for the INbreast dataset. Finally, we integrate the proposed classifier into a fully automated breast mass detection and segmentation, which shows promising results.

**Comment by cgreene** [comment]
The authors aim to train CNNs effectively with few features. To accomplish this, they first pre-train a CNN towards existing hand-crafted features. They then perform subsequent fine tuning using malignant/benign status. CNN performance (in terms of accuracy) with handcrafted features is good whether in fully automated or manual framings (in manual, the mass bounding box identification and segmentation are performed manually). From an ROC curve point of view performance remains lower with the automated approach than the manual approach. Interestingly, using CNN features with an RF classifier outperformed a CNN. This is also relevant in #25 where similar strong performance was observed for an RF layered over NN features.

**Comment by cgreene** [comment]
discussed in latest commits

**Comment by zoupei123456** [comment]
Why not disclose your code

**Comment by evancofer** [comment]
@zoupei123456 
This repository is not for this paper. This repository is the deep-review, a collaboratively written review on deep learning in precision medicine. If you are looking for the code from the paper mentioned in this issue, I recommend contacting the paper's corresponding authors with the email they provide in the manuscript.
